Add examples for confidential VM, VMSS and OS Disk

myc2h6o opened the pull request:
Adding examples for [Confidential VM](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#security_encryption_type), [Confidential VMSS](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_virtual_machine_scale_set#security_encryption_type) and [Confidential OS Disk](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/managed_disk#security_type)

myc2h6o committed:
Add examples for confidential VM, VMSS and OS Disk

grayzu commented on quickstart/201-confidential-os-disk/provider.tf:
```diff
@@ -0,0 +1,14 @@
+terraform {
```
please rename your provider.tf files to providers.tf (with "s") to be consistent with documentation.

grayzu commented on quickstart/201-confidential-vm/variables.tf:
```diff
@@ -0,0 +1,15 @@
+
+variable "name_prefix" {
+  type        = string
+  default     = "tftest"
```
Should leave this default off and make this a required parameter.

grayzu commented on quickstart/201-confidential-os-disk/variables.tf:
```diff
@@ -0,0 +1,10 @@
+
+variable "name_prefix" {
+  type        = string
+  default     = "tftest"
```
Please leave this default value off and make this a required variable.

grayzu commented on quickstart/201-confidential-vm/main.tf:
```diff
@@ -0,0 +1,145 @@
+  name                = "${var.name_prefix}-vm"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  size                = "Standard_DC2as_v5"
```
Why are we using this VM size? I do not necessarily have an issue with it but wondering why it was chosen.

myc2h6o replied:
there is limited size which can be used for confidential VM/VMSS. To avoid confusion, I've added the document link in the comment in the update.

grayzu commented on quickstart/201-confidential-vmss/main.tf:
```diff
@@ -0,0 +1,55 @@
+  name                 = "${var.name_prefix}-vmss"
+  resource_group_name  = azurerm_resource_group.example.name
+  location             = azurerm_resource_group.example.location
+  sku                  = "Standard_DC2as_v5"
```
same

grayzu commented on quickstart/201-confidential-vmss/provider.tf:
```diff
@@ -0,0 +1,14 @@
+terraform {
```
same providers.tf

grayzu commented on quickstart/201-confidential-vmss/variables.tf:
```diff
@@ -0,0 +1,16 @@
+
+variable "name_prefix" {
+  type        = string
+  default     = "tftest"
```
same comment as above

grayzu reviewed (changes requested):
Looks great! Thanks for the PR.

Few minor tweaks.

myc2h6o committed:
resolve comments

myc2h6o commented:
Hi @grayzu thanks for reviewing the pr! I've resolved the comment, can you take another look?
To summary the update:
- renamed provider.tf to provider**s**.tf
- removed the default value for `name_prefix` and reorder the variables
- added doc link in comments for vm size and vmss sku `Standard_DC2as_v5`, to include the available sizes to use confidential vm

grayzu commented:
Thanks @myc2h6o . Looks good to me now.
